-1

Tested at: d49909bb0f9cbd354dc1dcf175e1257cb3b9c95e

CMSSW: CMSSW_11_1_X_2020-02-10-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-98755c/4584/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>                 ^
/cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/external/pythia8/243/include/Pythia8/SpaceShower.h:146:18: note: hidden overloaded virtual function 'Pythia8::SpaceShower::getSplittingProb' declared here: different number of parameters (5 vs 7)
  virtual double getSplittingProb( const Event& , int , int , int , string )
                 ^
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-10-1100/src/GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc:61:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-02-10-1100/src/GeneratorInterface/Core/interface/ConcurrentHadronizerFilter.h:139:20: error: 'callWhenNewProductsRegistered' following the 'template' keyword does not refer to a template
    this->template callWhenNewProductsRegistered([ptrThis](BranchDescription const& iBD) {
                   ^
/cvmfs/cms-ib.cern.ch/nweek-02615/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/lib/gcc/x86_64-unknown-linux-gnu/8.3.1/../../../../include/c++/8.3.1/bits/unique_ptr.h:831:34: note: in instantiation of member function 'edm::ConcurrentHadronizerFilter<Pythia8Hadronizer, gen::ConcurrentExternalDecayDriver>::ConcurrentHadronizerFilter' requested here
    { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
                                 ^
</pre>

--- a/GeneratorInterface/Core/interface/ConcurrentHadronizerFilter.h
+++ b/GeneratorInterface/Core/interface/ConcurrentHadronizerFilter.h
@@ -136,7 +136,7 @@ namespace edm {
         nAttempts_(1),
         hasFilter_(ps.exists("HepMCFilter")) {
     auto ptrThis = this;
-    this->template callWhenNewProductsRegistered([ptrThis](BranchDescription const& iBD) {
+    this->callWhenNewProductsRegistered([ptrThis](BranchDescription const& iBD) {
       //this is called each time a module registers that it will produce a LHERunInfoProduct
       if (iBD.unwrappedTypeID() == edm::TypeID(typeid(LHERunInfoProduct)) && iBD.branchType() == InRun) {
         ++(ptrThis->counterRunInfoProducts_);

